go/vt/vtgate/evalengine/fn_string.go

```diff
@@ -253,6 +265,99 @@ func (call *builtinASCII) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.Int64, Col: collationNumeric, Flag: nullableFlags(str.Flag)}, nil
 }
 
+func reverse(in *evalBytes) []byte {
+	cs := colldata.Lookup(in.col.Collation).Charset()
+	b := in.bytes
+
+	out, cur := make([]byte, len(b)), len(b)
+	for _, size := cs.DecodeRune(b); size > 0; _, size = cs.DecodeRune(b) {
+		copy(out[cur-size:cur], b[:size])
+		b = b[size:]
+		cur = cur - size
+	}
```

Comment:
I'd rename `cur` here to `end` since that's a better name I think and I think it can be slightly further simplified. What about something like this:

```go
	out, end := make([]byte, len(b)), len(b)
	for len(b) > 0 {
		_, size := cs.DecodeRune(b)
		copy(out[end-size:end], b[:size])
		b = b[size:]
		end -= size
	}
```

I think it makes the loop condition easier to read this way. Tend to use the `for` with 3 parts only really for the most basic loop iteration increment to make it easier to understand.

Reply:
true, it does look more readable. Thanks, done.